projects/plugins/boost/app/modules/cache/Boost_Cache.php

```diff
@@ -254,4 +272,208 @@ public function is_backend() {
+			return false;
+		}
+		return true;
+	}
+
+	/*
+	 * Delete the cache for the front page and paged archives.
+	 * This is called when a post is edited, deleted, or published.
+	 *
+	 * @param int $post_id - The id of the post that was edited, deleted, or published.
+	 */
+	protected function delete_front_page_cache( $post ) {
+		$front_page_id = get_option( 'show_on_front' ); // posts page
+		if ( $front_page_id === 'page' ) {
+			$front_page_id = get_option( 'page_on_front' ); // static page
+			if ( $front_page_id === $post->ID ) {
+				$this->delete_cache_for_post( $post->ID );
+			}
+		} else {
+			$this->storage->invalidate_home_page();
+			error_log( 'delete front page cache' ); // phpcs:ignore WordPress.PHP.DevelopmentFunctions.error_log_error_log
+		}
+	}
+
+	/*
+	 * Delete the cache for the post if the comment transitioned from one state to another.
+	 *
+	 * @param string $new_status - The new status of the comment.
+	 * @param string $old_status - The old status of the comment.
+	 * @param WP_Comment $comment - The comment that transitioned.
+	 */
+	public function delete_on_comment_transition( $new_status, $old_status, $comment ) {
+		if ( $new_status === $old_status ) {
+			return;
+		}
+		$post = get_post( $comment->comment_post_ID );
+		$this->delete_cache_for_post( $post );
+	}
+
+	/*
+	 * After editing a comment, delete the cache for the post if the comment is approved.
+	 * If changing state and editing, both actions will be called, but the cache will only be deleted once.
+	 *
+	 * @param int $comment_id - The id of the comment.
+	 * @param array $commentdata - The comment data.
+	 */
+	public function delete_on_comment_edit( $comment_id, $commentdata ) {
+		$post = get_post( $commentdata['comment_post_ID'] );
+
+		if ( (int) $commentdata['comment_approved'] === 1 ) {
+			$this->delete_cache_for_post( $post );
+		}
+	}
+
+	/*
+	 * After a comment is posted, delete the cache for the post if the comment is approved.
+	 * If the comment is not approved, only delete the cache for this post for this visitor.
+	 *
+	 * @param int $comment_id - The id of the comment.
+	 * @param int $comment_approved - The approval status of the comment.
+	 * @param array $commentdata - The comment data.
+	 */
+	public function delete_on_comment_post( $comment_id, $comment_approved, $commentdata ) {
+		$post = get_post( $commentdata['comment_post_ID'] );
+
+		/*
+		 * If a comment is not approved, we only need to delete the cache for
+		 * this post for this visitor so the unmoderated comment is shown to them.
+		 */
+		if ( $comment_approved !== 1 ) {
+			$this->storage->invalidate_single_visitor( $this->normalize_request_uri( get_permalink( $post->ID ) ), $this->request_parameters );
+			return;
+		}
+
+		$this->delete_cache_for_post( $post );
+	}
+
+	/*
+	 * Returns true if the post is published or private.
+	 *
+	 * @param string $status - The status of the post.
+	 * @return bool
+	 */
+	private function is_published( $status ) {
```

Comment:
Should this go in `Boost_Cache_Utils`?

Reply:
It probably doesn't because the only code that checks the status like this will be the post transition function.